src/sentry/static/sentry/app/components/groupTombstones.jsx

```diff
@@ -0,0 +1,115 @@
+import React from 'react';
+
+import AlertActions from '../actions/alertActions';
+
+import LoadingError from '../components/loadingError';
+import LinkWithConfirmation from '../components/linkWithConfirmation';
+
+import ApiMixin from '../mixins/apiMixin';
+
+import {t} from '../locale';
+
+const GroupTombstones = React.createClass({
+  propTypes: {
+    orgId: React.PropTypes.string.isRequired,
+    projectId: React.PropTypes.string.isRequired,
+    tombstones: React.PropTypes.array.isRequired,
+    tombstoneError: React.PropTypes.bool.isRequired
+  },
+
+  mixins: [ApiMixin],
+
+  undiscard(tombstoneId) {
+    // TODO (kt): update this when you scope the API endpoint to the project
+    let path = `/tombstone/${tombstoneId}/`;
+    this.api.request(path, {
+      method: 'DELETE',
+      success: data => {
+        AlertActions.addAlert({
+          message: t('Events similar to these will no longer be filtered'),
+          type: 'success'
+        });
+      },
+      error: () => {
+        AlertActions.addAlert({
+          message: t('We were unable to discard this group'),
+          type: 'error'
+        });
+      }
+    });
+    this.fetchData();
+  },
+
+  renderEmpty() {
+    return <div className="box empty">{t('None')}</div>;
+  },
+
+  render() {
+    if (this.props.tombstoneError) return <LoadingError />;
+
+    let {tombstones} = this.props;
+    return (
+      <div>
+        <div className="row" style={{paddingTop: 10}}>
+          <div className="col-sm-8">
+            <h5>{t('Discarded Groups')}</h5>
+            {tombstones.length
+              ? <ul className="group-list">
+                  {tombstones.map(data => {
+                    return (
+                      <li
```

Comment:
This is pretty nested. I'd probably [extract this into a functional component](https://facebook.github.io/react/docs/components-and-props.html#extracting-components) called `GroupTombstoneRow`, declared above at the top of this file.